packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -1311,14 +1314,23 @@ export function makeWallet({
     return issuerManager;
   }
 
-  const offerResultStore = makeWeakStore('invitationHandle');
-
   async function saveOfferResult(invitationHandle, offerResult) {
-    offerResultStore.init(invitationHandle, offerResult);
+    invitationHandleToOfferResult.init(invitationHandle, offerResult);
   }
 
   async function getOfferResult(invitationHandle) {
-    return offerResultStore.get(invitationHandle);
+    return invitationHandleToOfferResult.get(invitationHandle);
+  }
+
+  async function getUINotifier(rawId, dappOrigin = 'unknown') {
+    const id = makeId(dappOrigin, rawId);
+    const offerResult = idToOfferResult.get(id);
+    assert(
+      passStyleOf(offerResult) === 'copyRecord',
+      `offerResult must be a record to have a uiNotifier`,
+    );
+    assert(offerResult.uiNotifier);
```

Comment:
Add a failure description string to this?  `offerResult does not have a uiNotifier`?

Reply:
good idea